lib/private/Security/Crypto.php

```diff
@@ -90,16 +82,19 @@ public function encrypt(string $plaintext, string $password = ''): string {
 		if ($password === '') {
 			$password = $this->config->getSystemValue('secret');
 		}
-		$this->cipher->setPassword($password);
+		
+		$keyMaterial = hash_hkdf('sha512', $password);
```

Comment:
```suggestion
		$keyMaterial = hash_hkdf('sha512', $password, 32);
```

I'm not 100% sure. The [documentation](https://www.php.net/manual/en/function.hash-hkdf.php
) says "Desired output length in bytes" :man_shrugging:

Reply:
@kesselb The documentation says `hash_hkdf` with a `$length` of `0` will use the provided hash function length. Also, 32 bytes is not enough. If anything, we'll explicitly ask for 64 bytes.